ch04packaging/04documentation.ipynb

```diff
@@ -393,14 +379,115 @@
     "\n"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Doctest - testing your documentation is up to date"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`doctest` is a module included in the standard library. It runs all the code within the docstrings and checks whether the output is what it's claimed on the documentation.\n",
+    "\n",
+    "Let's add an example to our greeting function and check it with `doctest`."
+   ]
+  },
   {
    "cell_type": "code",
-   "execution_count": null,
-   "metadata": {
-    "collapsed": true
-   },
-   "outputs": [],
-   "source": []
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Overwriting greetings/greetings/greeter.py\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%writefile greetings/greetings/greeter.py\n",
+    "def greet(personal, family, title=\"\", polite=False):\n",
+    "    \"\"\" Generate a greeting string for a person.\n",
+    "\n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    personal: str\n",
+    "        A given name, such as Will or Jean-Luc\n",
+    "    family: str\n",
+    "        A family name, such as Riker or Picard\n",
+    "    title: str\n",
+    "        An optional title, such as Captain or Reverend\n",
+    "    polite: bool\n",
+    "        True for a formal greeting, False for informal.\n",
+    "\n",
+    "    Returns\n",
+    "    -------\n",
+    "    string\n",
+    "        An appropriate greeting\n",
+    "        \n",
+    "    Examples\n",
+    "    --------\n",
+    "    >>> from greetings.greeter import greet\n",
+    "    >>> greet(\"Terry\", \"Jones\")\n",
+    "    'Hey, Terry Jones.\n",
+    "    \"\"\"\n",
+    "\n",
+    "    greeting= \"How do you do, \" if polite else \"Hey, \"\n",
+    "    if title:\n",
+    "        greeting += f\"{title} \"\n",
+    "\n",
+    "    greeting += f\"{personal} {family}.\"\n",
+    "    return greeting"
+   ]
+  },
```

Comment:
Maybe here add a markdown cell that makes it clear we have made a mistake in the docstring? So that it's clear that the part below fails intentionally.